### DISCUSSION TITLE
Streaming Large Files

### DISCUSSION THREAD

**Discussion by metawrap-dev** [ohmjs/ohm#468] [Ideas]
Great project! I'm using it in something very interesting. 

I'm wondering if it would be possible to modify InputStream and some of the semantics around checking eof, and stream data in chunks into the parser from an actual stream?

The goal is that the whole source file is never loaded at once into memory,

Use cases: Parsing an infinite network stream and parsing very large files.

**Comment by pdubroy** [comment]
Hi! Sorry it's taken so long to get around to answering this. It's a good idea, I've pulled it into an issue: #516
